src/main/java/org/gridsuite/study/server/service/shortcircuit/ShortCircuitService.java

```diff
@@ -116,51 +117,70 @@ private String getShortCircuitAnalysisResultResourcePath(UUID nodeUuid, Shortcir
+    private String getShortCircuitAnalysisResultsPageResourcePath(UUID nodeUuid, ShortcircuitAnalysisType type) {
+        String resultPath = getShortCircuitAnalysisResultResourcePath(nodeUuid, type);
         if (resultPath == null) {
             return null;
         }
-        return UriComponentsBuilder.fromPath(resultPath + "/fault_results/paged").toUriString();
+        if (type == ShortcircuitAnalysisType.ALL_BUSES) {
+            resultPath += "/fault_results";
+        } else if (type == ShortcircuitAnalysisType.ONE_BUS) {
+            resultPath += "/feeder_results";
+        } else {
+            throw new StudyException(SHORT_CIRCUIT_ANALYSIS_BAD_TYPE);
+        }
+        return resultPath + "/paged";
     }
 
     public String getShortCircuitAnalysisResult(UUID nodeUuid, String mode, ShortcircuitAnalysisType type) {
-        // For ONE_BUS results, we always want full results mode
-        String overridedMode = type == ShortcircuitAnalysisType.ONE_BUS ? "FULL" : mode;
-        String params = "?mode=" + overridedMode;
         String resultPath = getShortCircuitAnalysisResultResourcePath(nodeUuid, type);
         if (resultPath == null) {
             return null;
         }
-        return getShortCircuitAnalysisResource(resultPath + params);
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(shortCircuitServerBaseUri + resultPath)
+                .queryParam("mode", mode);
+
+        return getShortCircuitAnalysisResource(builder.build().toUri());
     }
 
-    public String getShortCircuitAnalysisFaultResultsPage(UUID nodeUuid, String mode, Pageable pageable) {
-        StringBuilder paramsBuilder = new StringBuilder();
-        paramsBuilder.append("?mode=" + mode + "&page=" + pageable.getPageNumber() + "&size=" + pageable.getPageSize());
+    public String getShortCircuitAnalysisResultsPage(UUID nodeUuid, String mode, ShortcircuitAnalysisType type, String filters, Pageable pageable) {
+        String resultsPath = getShortCircuitAnalysisResultsPageResourcePath(nodeUuid, type);
+        if (resultsPath == null) {
+            return null;
+        }
 
-        for (Sort.Order order : pageable.getSort()) {
-            paramsBuilder.append("&sort=" + order.getProperty() + "," + order.getDirection());
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(shortCircuitServerBaseUri + resultsPath)
+                .queryParam("mode", mode)
+                .queryParam("page", pageable.getPageNumber())
+                .queryParam("size", pageable.getPageSize());
+
+        if (filters != null && !filters.isEmpty()) {
+            builder.queryParam("filters", filters);
         }
-        String faultResultsPath = getShortCircuitAnalysisFaultResultsResourcePath(nodeUuid);
-        if (faultResultsPath == null) {
-            return null;
+
+        for (Sort.Order order : pageable.getSort()) {
```

Comment:
We should add a function which take a `Pageable` and return query params because it will be used on other places where pagination is needed

Reply:
Good idea, done